### PR TITLE
Child disability allowance

### PR DESCRIPTION
**Pull request by asgharkazi**
* Tax and benefit system evolution. | Technical improvement. | Crash fix. | Minor change.
* Impacted periods: all. | until DD/MM/YYYY. | from DD/MM/YYYY.
* Impacted areas: `path/to/file/containing/impacted/variables`
* Details:
  - New feature or new behaviour description
  - Cases for which an error was noticed

- - - -

These changes:
- Impact the OpenFisca-Aotearoa public API (for instance renaming or removing a variable)
- Add a new feature (for instance adding a variable)
- Fix or improve an already existing calculation.

### CODE REVIEW

File: openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+    def formula(persons, period, parameters):
+        # Do they have a partner
+        no_partners = (persons("person_has_partner", period) == 0)
+        not_supported = (persons("disability_allowance__is_adequately_supported_by_partner", period) == 0)
+        # no partner, OR not supported by partner
+        return no_partners + not_supported
+
+
+class disability_allowance__current_income(variables.Variable):
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py:333]: What is the difference with pure _income_?

File: openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    value_type = int
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "How much does the person earn per week? (including Superannuation income or Veteran's Pension?)"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+
+class disability_allowance__family_income(variables.Variable): # Copied from social security job_seeker formulas
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py:341]: Could the scope of this definition be broader? Prefixing should be the exception, is family income defined differently for disability allowance than for jobseeker support, or other benefits, or could it be generalised?

File: openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+    value_type = bool
+    default_value = True
+    entity = entities.Person
+    label = "Meets the sole parent support test for not being in a relationship"
+    definition_period = periods.WEEK
+    reference = "https://www.workandincome.govt.nz/map/income-support/main-benefits/sole-parent-support/qualifications.html"
+
+    def formula(persons, period, parameters):
+        # Do they have a partner
+        no_partners = (persons("person_has_partner", period) == 0)
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py:327]: Couldn't we rationalise here the use of _person_has_parner_ and _in_a_relartionship_?

File: openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+    value_type = bool
+    default_value = False
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "Does the person earn below the Disability Allowance Income Limit as per clause 14?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula_2018_11_26(persons, period, parameters):
+        sole_parent = persons("is_sole_parent", period) * \
+        persons("disability_allowance__sole_parent_meets_relationship_qualification", period)
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py:296]: Same as in the other PR, you have to sum here.

File: openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+    default_value = False
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "Does the person earn below the Disability Allowance Income Limit as per clause 13?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula_2018_11_26(persons, period, parameters):
+        sole_parent = persons("is_sole_parent", period) * \
+        persons("disability_allowance__sole_parent_meets_relationship_qualification", period)
+        only_one_child = (persons("social_security__dependent_children", period) == 1)
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py:280]: You hacve to sum here.

File: openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+class disability_allowance__income_limit_clause_12(variables.Variable):
+    value_type = bool
+    default_value = False
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "Does the person earn below the Disability Allowance Income Limit as per clause 12?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula_2018_11_26(persons, period, parameters):
+        in_relationship = persons("person_has_partner", period) #refactor this to person_has_partner
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py:257]: If you rebase from main you'll see that I rationalised the usde of this varfiabvle

**Review by bonjourmauko — CHANGES_REQUESTED** [review]
Looks good. As it depends on the other one, please re-send me a review request once the other one qgets merged.

**Review by MarkCalvert — APPROVED** [review]
Will add requested changes in a future PR

**Pull request closed without merging**